backend/src/scripts/migrations/4.30/moveTSDataToTSSettings.js

```diff
@@ -0,0 +1,87 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { v5Path } = require('../../../interop');
+
+const { find, listDatabases, updateMany, updateOne } = require(`${v5Path}/handler/db`);
+const { ADD_ONS } = require(`${v5Path}/models/teamspaces.constants`);
+const { USERS_DB_NAME } = require(`${v5Path}/models/users.constants`);
+const { deleteIfUndefined } = require(`${v5Path}/utils/helper/objects`);
+const { logger } = require(`${v5Path}/utils/logger`);
+
+const addOnFields = Object.values(ADD_ONS);
+
+const migrateTeamspaceData = async (user, customData) => {
+	const { addOns, billing: { subscriptions }, permissions, ...flags } = customData;
+	const tsSettingsUpdate = deleteIfUndefined({
+		subscriptions,
+		permissions,
+	});
+
+	// addOns
+	if (addOns) {
+		tsSettingsUpdate.addOns = { ...addOns, ...flags };
+	}
+
+	if (Object.keys(tsSettingsUpdate).length > 0) {
+		try {
+			await updateOne(user, 'teamspace', {}, { $set: tsSettingsUpdate });
+			return user;
+		} catch (err) {
+			logger.logError(`\t-Update teamspace settings for ${user} unsuccessful: ${err}`);
+		}
+	}
+
+	return undefined;
+};
+
+const run = async () => {
+	const oldFields = {
+		'customData.permissions': 1,
+		'customData.billing.subscriptions': 1,
+		'customData.addOns': 1,
+	};
+	addOnFields.forEach((addOn) => {
+		oldFields[`customData.${addOn}`] = 1;
+	});
+
+	const query = { $or: Object.keys(oldFields).map((key) => ({ [key]: { $exists: true } })) };
+
+	const dbs = (await listDatabases()).map(({ name }) => name);
```

Comment:
Since you're looping through the array here anyway, you're better off putting the list of dbs in a set or a dictionary, so you can do a more efficient search than doing `dbs.includes` at L74